src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -341,17 +322,16 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                     catch (SearchParameterNotSupportedException)
                     {
                         sortingsValid = false;
-                        _contextAccessor.RequestContext?.BundleIssues.Add(new OperationOutcomeIssue(
-                            OperationOutcomeConstants.IssueSeverity.Warning,
-                            OperationOutcomeConstants.IssueType.NotSupported,
-                            string.Format(CultureInfo.InvariantCulture, Core.Resources.SearchParameterNotSupported, sorting.Item1, string.Join(", ", resourceTypesString))));
+                        searchSortErrors.Add(string.Format(CultureInfo.InvariantCulture, Core.Resources.SearchParameterNotSupported, sorting.Item1, string.Join(", ", resourceTypesString)));
```

Comment:
Should we be handing multiple bad sort parameters the same way we handle multiple bad search parameters? Instead of one error per bad parameters we have one error that concatenates all the bad parameters into one message.

Reply:
https://github.com/microsoft/fhir-server/blob/410600f34903bd7a9ff9a9ef525d4d5b1327fbdb/src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs#L306-L325

Current code behaves differently for errors and warnings, shouldn't it behave the same for all the cases? For errors it concatenates and for warnings it creates separate OperationOutcomeIssue for each incorrect parameter. DO you really think we should just copy the current pattern or fix the existing code as well?

Reply:
My opinion is it should behave same for errors and warnings.

Reply:
I agree with that. I don't know what the original thoughts were on handling them differently. My preference would be for one operation outcome, instead of multiple. @brendankowitz What are your thoughts?

Reply:
We may end up with multiple issues anyway (can someone confirm?) i.e. unsupported modifiers, search parameter types, truncated includes, etc. 
We will only have one OperationOutcome object, but it can contain multiple issues, I think one issue per problem seems ok

Reply:
Fixed preexisting code. Now all errors/warnings/sort/search are handled in the same way. No concatenation, each issue gets it's own OperationOutcomeIssue instance. Reason is because multiple errors strings of same type could be created in very different places like search code and then again sort code. Concatenating would mean that already generated error strings would have to be parsed so they can be grouped by error type, then parameters in the strings would have to be extracted and then concatenated and then put into a new error string of same error type,